src/common_labsoft/components/http_client.clj

```diff
@@ -28,36 +29,44 @@
 (defn resolve-token
   "Receives a token. If its valid, returns it. If its not, generate a valid token and return it."
   [token-component token service-name service-password]
-  (let [auth-payload {:auth/service-name service-name :auth/auth-password service-password}
+  (let [auth-payload {:auth/service service-name :auth/password service-password}
         config (:config token-component)
         services (protocols.config/get! config :services)]
     (if (protocols.token/verify token-component token)
       token
       (-> (render-route services :auth :service-token)
           (client/post {:form-params auth-payload :content-type :json})
-          (:body)
+          :body
           (cheshire/parse-string true)
-          (:token/jwt)))))
+          :token/jwt))))
 
 ;; Transform methods: receives and returns request's data.
 (defn transform-url
   "Generates a url if the data was provided with host/endpoint or keyworded url. Applies replace-map"
-  [data hosts]
-  (let [service (:host data)
-        endpoint (:endpoint data)
-        replace-map (:replace-map data {})
-        url (:url data)]
-    (if (keyword? url)
-      (let [url (render-keyworded-url hosts url (:replace-map data {}))]
-        (merge (dissoc data :replace-map) {:url url}))
-      (if (and (some? service) (some? endpoint))
-        (merge (dissoc data :replace-map :endpoint :host) {:url (render-route hosts service endpoint replace-map)})
-        data))))
+  [{:keys [host endpoint replace-map url] :as data} hosts]
+  (if (keyword? url)
+    (let [url (render-keyworded-url hosts url (or replace-map {}))]
+      (merge (dissoc data :replace-map) {:url url}))
+    (if (and (some? host) (some? endpoint))
+      (merge (dissoc data :replace-map :endpoint :host) {:url (render-route hosts host endpoint (or replace-map {}))})
+      data)))
 
 (defn transform-method
   "Transforms request's method"
-  [data method]
-  (merge data {:method method}))
+  [method data]
+  (assoc data :method method))
+
+(defn internalize-response
+  [req resp]
+  (let [json-or-edn (if (= (get-in resp [:headers :content-type]) :application/json)
```

Comment:
I don't know if the headers map is going to come with keywords keys, probably not, and the value absolutely is going to be a string, just double check with clj-http doc